sematic/db/migrations/20221215212459_external_resources.sql

```diff
@@ -0,0 +1,25 @@
+-- migrate:up
+
+CREATE TABLE external_resources (
+    id character(32) NOT NULL,
+    resource_state TEXT NOT NULL,
+    status_message TEXT NOT NULL,
+    last_updated_epoch_seconds int8 NOT NULL,
```

Comment:
The rest of the DB timestamp columns use the `timestamp WITH time zone` type. Why `int8` here?

Reply:
Either way, there should be `created_at` and `updated_at` fields on this table that are automatically set by SQLAlchemy.

```python
    created_at: datetime.datetime = Column(
        types.DateTime(), nullable=False, default=datetime.datetime.utcnow
    )
    updated_at: datetime.datetime = Column(
        types.DateTime(),
        nullable=False,
        default=datetime.datetime.utcnow,
        onupdate=datetime.datetime.utcnow,
    )
```

Reply:
Added `created_at` and `updated_at` as DateTimes for consistency with other tables. I advocate that we keep `last_update_epoch_time` as-is because  `ResourceStatus` is not intrinsically a DB object. Timezones are a pain in the butt to deal with properly, and I prefer the simplicity and lack of ambiguity around epoch timestamps.

Reply:
`last_update_epoch_time` is a part of the dataclass for `ExternalResource`, and is the timestamp of the last change to said dataclass. It is also the only update information accessible in most of the framework, since the rest of the framework works with the dataclass. `updated_at` is the last time the DB record was saved, which may be distinct from when the dataclass was updated (certainly will be by at least some small amount of time). The difference between these might be useful for debugging situations (ex: the DB record was updated 5 minutes ago, but the contents of the dataclass have been unchanged for hours--this means that we've been saving the record a bunch but the actual remote resource being represented has been in a static state). It will also be useful to any framework level code that needs to know the last time the remote object info was refreshed.